### ISSUE TITLE
CUBEJS_SCHEDULED_REFRESH_TIMER=false still starts CubeJs as scheduler instance

### ISSUE THREAD

**Issue by craigbeck** [cube-js/cube.js#1972]
**Describe the bug**

Starting CubeJs (as express app, not official docker image) when explicitly specifying `CUBEJS_SCHEDULED_REFRESH_TIMER=false` still starts scheduler instance with 30000ms schedule

**To Reproduce**
1. `CUBEJS_SCHEDULED_REFRESH_TIMER=false yarn start`
2. wait ~30 seconds
3. watch scheduled queries in logs

**Expected behavior**
No scheduler started

**Screenshots**
```
$ CUBEJS_SCHEDULED_REFRESH_TIMER=false yarn start
yarn run v1.22.10
$ CUBEJS_DB_BQ_KEY_FILE=${CUBEJS_DB_BQ_KEY_FILE:-./client_key.json} ALLOW_UNSECURE_HTTP=1 nodemon -w ./schema index.js
[nodemon] 2.0.7
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): schema/**/*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `node index.js`
...
🔓 Authentication checks are disabled in developer mode. Please use NODE_ENV=production to enable it.
🦅 Dev environment available at http://localhost:4000
(node:82019) Warning: Accessing non-existent property 'INVALID_ALT_NUMBER' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
(node:82019) Warning: Accessing non-existent property 'INVALID_ALT_NUMBER' of module exports inside circular dependency
🚀 Cube.js server v0.25.32 is listening on 4000
Performing query: scheduler-c2927158-fc46-40a1-8448-1306541493bf 
driverFactory default
Performing query: scheduler-c2927158-fc46-40a1-8448-1306541493bf 
driverFactory postgres
Performing query: scheduler-c2927158-fc46-40a1-8448-1306541493bf 
Performing query: scheduler-c2927158-fc46-40a1-8448-1306541493bf 
driverFactory postgres
Executing SQL: scheduler-c2927158-fc46-40a1-8448-1306541493bf 
--
  SELECT FLOOR((UNIX_SECONDS(CURRENT_TIMESTAMP())) / 120)
--
Executing SQL: scheduler-c2927158-fc46-40a1-8448-1306541493bf 
```

**Version:**
`@cubejs-backend/bigquery-driver@0.25.32`
 `@cubejs-backend/postgres-driver@0.25.32`
 `@cubejs-backend/server@0.25.32`

**Additional context**
Inserted the following locally at L#149 of server.js (https://github.com/cube-js/cube.js/blob/master/packages/cubejs-server-core/src/core/server.ts#L186)
```
  console.log('****** scheduledRefreshTimer', scheduledRefreshTimer, '******');
  console.log('options', options);
  console.log(process.env);
```
and see output like:
```
****** scheduledRefreshTimer 30000 ******
options {
  dbType: [Function: dbType],
  externalDbType: undefined,
  devServer: true,
  driverFactory: [AsyncFunction: driverFactory],
  dialectFactory: [Function: dialectFactory],
  externalDriverFactory: undefined,
  externalDialectFactory: [Function: externalDialectFactory],
  apiSecret: <redacted>
  telemetry: true,
  scheduledRefreshTimeZones: undefined,
  scheduledRefreshContexts: [AsyncFunction: scheduledRefreshContexts],
  basePath: '/cubejs-api',
  dashboardAppPath: 'dashboard-app',
  dashboardAppPort: 3000,
  scheduledRefreshConcurrency: NaN,
  preAggregationsSchema: 'stb_pre_aggregations',
  schemaPath: 'schema',
  checkAuthMiddleware: [Function: checkAuthMiddleware],
  queryTransformer: [Function: queryTransformer]
}
```
and
```
  CUBEJS_SCHEDULED_REFRESH_TIMER: 'false',
```
from the `process.env` output

> Note: does not schedule anything if env var is not set

**Comment by rongfengliang** [comment]
@craigbeck  this is a bug  i do some fix  https://github.com/cube-js/cube.js/pull/1973

**Comment by craigbeck** [comment]
found another issue #1985 and dug into code some. Looks like problem is in `detectScheduledRefreshTimer` https://github.com/cube-js/cube.js/blob/7d15099462e60cb666bd9342580583ddf325c2ab/packages/cubejs-server-core/src/core/server.ts#L351

1. `scheduledRefreshTimer` argument is `undefined` it matches this block: https://github.com/cube-js/cube.js/blob/7d15099462e60cb666bd9342580583ddf325c2ab/packages/cubejs-server-core/src/core/server.ts#L363-L365
2. this then sets `scheduledRefreshTimer = process.env.NODE_ENV !== 'production'` (e.g. running in dev locally) https://github.com/cube-js/cube.js/blob/7d15099462e60cb666bd9342580583ddf325c2ab/packages/cubejs-server-core/src/core/server.ts#L363-L365
3. and the following block then matches that the type is boolean, and sets `30000ms` value for timer https://github.com/cube-js/cube.js/blob/7d15099462e60cb666bd9342580583ddf325c2ab/packages/cubejs-server-core/src/core/server.ts#L367-L369

**Comment by craigbeck** [comment]
It does however look like tests _expect_ this behavior
https://github.com/cube-js/cube.js/blob/7d15099462e60cb666bd9342580583ddf325c2ab/packages/cubejs-server-core/test/unit/index.test.ts#L180

where `value=undefined`, and expected result is `30000`, and implicit default of `NODE_ENV != production`

**Comment by rongfengliang** [comment]
@craigbeck  

if scheduledRefreshTimer is undefined and 

```code
scheduledRefreshTimer == null is true 
```

maybe 

```code

scheduledRefreshTimer ===  null this is right way for check  scheduledRefreshTimer
```

**Comment by ovr** [comment]
Fix was release in `v0.26.11`. Please us 

To background scheduler, you can use:

```
CUBEJS_SCHEDULED_REFRESH_TIMER=false
# or
CUBEJS_SCHEDULED_REFRESH=false
# or pass in configuration
scheduledRefreshTimer: false
```

Thanks
